docs/RELEASE.md

```diff
@@ -0,0 +1,66 @@
+# Managing Releases
+
+
+## Handling patch releases
+
+### When should I create a patch release?
+Patch releases are exclusively for bugs and and security fixes. Patch releases can have minor _enhancements_ to 
+existing features but really should not affect the API in any noticeable way (unless it is adding an argument). For any
+functionality changes or new endpoints, please create a minor or major release.
+
+A patch release should increment the third digit e.g. `0.6.9 -> 0.6.10`. This indicates to users that the release 
+
+
+### Creating a patch release
+
+When looking through the branches on the main repo, you will notice a series of `release` branches labeled
+`release-<major version>.<minor version>`. So if we are currently working on the 0.6 release, the branch should be labeled 
+`release-0.6`. Before creating a minor release please check the [milestones](https://github.com/astro-projects/astro/milestones)
+page to ensure that all relevent bug fixes have been PRed and merged. 
+
+Once you merge all expected fixes, the next step is to ensure that all fixes have been cherry-picked to the release branch from main (please use `git cherry-pick -x <commit id>` to retain
```

Comment:
@dimberman, since we are referring to only commits that are fixes/security fix, would it be worth to add commits with certain category markers to indicate if they are normal dev commit or fix/security fix? or we gonna infer it from associated ticket?